### DISCUSSION TITLE
math/rand/v2

### DISCUSSION THREAD

**Discussion by brianvoe** [brianvoe/gofakeit#332] [General]
Looks like they are making changes to the math/rand package and are releasing a v2. https://github.com/golang/go/discussions/60751

Just trying to layout the side effect if Gofakeit were to upgrade to math/rand/v2

1-8. I dont see rmuch issue here as most of these are just minor updates. Gofakeit uses its own global rand so its not like i have to update too much having to do with initializing a new rand source.

9. Use [Lemire's algorithm](https://lemire.me/blog/2016/06/27/a-fast-alternative-to-the-modulo-reduction). This looks like it will bring a decent performance update. Most likely i will update the default to this. Will probably need to update all the examples and tests though :(

10. Add new PCG source

I most likely will have to create a new full version bump. Might take the opportunity to do some breaking clean ups.

Let me know if you see any issues with upgrading. Should i even upgrade at all?

**Comment by ChristoWolf** [comment]
I think it would be good 😊
